src/prefect/utilities/context.py

```diff
@@ -93,11 +93,15 @@ def __call__(self, *args: MutableMapping, **kwargs: Any) -> Iterator["Context"]:
         """
         previous_context = self.copy()
         try:
-            self.update(*args, **kwargs)
+            new_context = Context(*args, **kwargs)
+            if "config" in new_context:
+                new_context["config"] = as_nested_dict(new_context["config"])
```

Comment:
I've come to regret making DotDict the default class for a function called `as_nested_dict`, so my penance is to make it explicit.

```suggestion
                new_context["config"] = as_nested_dict(new_context["config"], dct_class=Config)
```

Reply:
Actually, this `dct_class` should be `Config` not `DotDict`. Updating my suggested code but not sure if that class is imported in this file.

Reply:
Updated